src/test/scala/firrtlTests/AnnotationTests.scala

```diff
@@ -74,4 +57,43 @@ class AnnotationTests extends AnnotationSpec with Matchers {
     val ta = Annotation(cName, classOf[Transform], "")
     execute(w, getAMap(ta), input, ta)
   }
+
+  "Annotations" should "be readable from file" in {
+    val annotationFile = new File("src/test/resources/annotations/SampleAnnotations.anno")
+    val annotationsYaml = io.Source.fromFile(annotationFile).getLines().mkString("\n").parseYaml
+    val annotationArray = annotationsYaml.convertTo[Array[Annotation]]
+    annotationArray.length should be (9)
+    annotationArray(0).targetString should be ("ModC")
+    annotationArray(7).transformClass should be ("firrtl.passes.InlineInstances")
+    val expectedValue = "TopOfDiamond\nWith\nSome new lines"
+    annotationArray(7).value should be (expectedValue)
+  }
+
+  "Round tripping annotations through text file" should "preserve annotations" in {
+    val annos: Array[Annotation] = Seq(
+      InlineAnnotation(CircuitName("fox")),
+      InlineAnnotation(ModuleName("dog", CircuitName("bear"))),
+      InlineAnnotation(ComponentName("chocolate", ModuleName("like", CircuitName("i")))),
+      PinAnnotation(CircuitName("Pinniped"), Seq("sea-lion", "monk-seal"))
+    ).toArray
+
+    val annoFile = new File("temp-anno")
+    val writer = new FileWriter(annoFile)
+    writer.write(annos.toYaml.prettyPrint)
+    writer.close()
+
+    val yaml = io.Source.fromFile(annoFile).getLines().mkString("\n").parseYaml
+    annoFile.delete()
+
+    val readAnnos = yaml.convertTo[Array[Annotation]]
+
+    annos.zip(readAnnos).foreach { case (beforeAnno, afterAnno) =>
+      beforeAnno.targetString should be (afterAnno.targetString)
+      beforeAnno.target should be (afterAnno.target)
+      beforeAnno.transformClass should be (afterAnno.transformClass)
+      beforeAnno.transform should be (afterAnno.transform)
+
+      beforeAnno should be (afterAnno)
+    }
+  }
```

Comment:
@jackkoenig not sure how that would happen unless someone hand crafts something with dot's in it, because otherwise these names are generated in chisel, but if this is a concern the constructors for the Named family should probably test for this.